test/html_to_json/Html_to_jsonTest.java

```diff
@@ -0,0 +1,167 @@
+    public static boolean isValidJSON(String test) {
+        try {
+            new JSONObject(test);
+        } catch (JSONException ex) {
+            // edited, to include @Arthur's comment
+            // e.g. in case JSONArray is valid as well...
+            try {
+                new JSONArray(test);
+            } catch (JSONException ex1) {
+                return false;
+            }
+        }
+        return true;
+    }
+    
+    @Test
+    public void checkConn() throws IOException {
+        //Tests connection to url
+        Connection conn = Jsoup.connect("https://github.com/egis/handbook/blob/master/Tech-Stack.md"); 
+        conn.get(); // would throw exception on error
+        assertTrue(true);
+    }
+    
+    
+    @Test
+    public void checkRedirects() {
+        //Tests for stable connection
+        boolean conn = true;
+        try {
+            Document doc = Jsoup.connect("https://github.com/egis/handbook/blob/master/Tech-Stack.md").get();
+        } catch (IOException e) {
+            assertTrue(e.getMessage().contains("Too many redirects"));
+            conn = false;
+        }
+        assertTrue(conn);
+    }
+    
+    @Test
+    public void checkCharset() throws IOException {
+        //Tests that we get back text/html in a "UFT8" character set
+        String url = "https://github.com/egis/handbook/blob/master/Tech-Stack.md";
+        Connection.Response res = Jsoup.connect(url).execute();
+        assertEquals("text/html; charset=utf-8", res.header("Content-Type"));
+        Document doc = res.parse();
+        assertTrue(doc.text().contains("Tech-Stack"));
+        assertEquals("UTF-8", res.charset());
+    }
+    
+    @Test
+    public void testMain() throws IOException {
+        //Tests our Main method in Html_to_json.class
+        // Declare boolean vars used throughout test to verify JSON format
+        boolean ans;
+        boolean val;
+        
+        //JSOUP connects to URL and gets HTML
+        String html = Jsoup.connect("https://github.com/egis/handbook/blob/master/Tech-Stack.md").get().html();
+        Document doc = Jsoup.parse(html);
+        
+        //Check that we have connected to the right page
+        assertEquals("handbook/Tech-Stack.md at master · egis/handbook · GitHub", doc.title());
+        
+        //Gets all <h2> tags
+        Element divID = doc.getElementById("readme");
+        Elements h2Tags = divID.getElementsByTag("h2");
+        
+        //Loop through each heading
+        for (int a = 0, b = h2Tags.size(); a < b; a++) {
+            int headNum = a+1;
+            Elements heading = h2Tags.eq(a).select("h2");
+            JSONObject jsonHeadObject = new JSONObject();
+            jsonHeadObject.put("Heading:"+headNum, heading.text());
+            
+            //Verify correct JSON conversion
+            ans = true;
+            val = isValidJSON(jsonHeadObject.toString());
+            
+            assertEquals(ans,val);
```

Comment:
You're creating your JSON object from scratch and adding only one value to it - by default it will always be valid. Your _jsonHeadObject.put_ would fail if the parameters you provided would've been invalid, but since you got to this line already, you're guaranteed that your JSON is valid.